### PR TITLE
Add final single part ops

### PR DESCRIPTION
**Pull request by sbailey-arm**
Added missing documentation for all single part operations (excluding copy - requires discussion on implementation for Parsec)

### CODE REVIEW

File: src/parsec_client/operations/psa_purge_key.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# PsaPurgeKey
```
Comment (hug-dev) [inline src/parsec_client/operations/psa_purge_key.md:1]: Similarly to the comment on your PR on `parsec-operations`, I think it is best to remove it from now.

File: src/parsec_client/operations/psa_sign_hash.md [hunk 2]
```diff
@@ -37,7 +37,7 @@ such signature algorithm: Raw PKCS#1 v1.5 signature.
 **Note:** To perform a hash-and-sign algorithm, the hash must be calculated before passing it to
 this function. This could be done with the operation PsaHashCompute or with a multi-part hash
 operation. Those operations are not yet implemented. Alternatively, to hash and sign a message in a
-single call, you could use PsaSignMessage (not yet implemented).
+single call, you could use PsaSignMessage.
```
Comment (hug-dev) [inline src/parsec_client/operations/psa_sign_hash.md:40]: Thanks for that 👍

**Review by hug-dev — COMMENTED** [review]
Thanks!! Everything look code for me, small comment about PurgeKey.

File: src/parsec_client/operations/psa_purge_key.md (hunk 1 above)
Comment (sbailey-arm) [reply src/parsec_client/operations/psa_purge_key.md:1]: 🧹

File: src/clients_api_coverage.md [hunk 3]
```diff
@@ -90,6 +95,27 @@ will be added here for any other such coverage report.
...
+| CbcNoPadding | ❌         |
+| CbcPkcs7     | ❌         |
+
+### MAC algorithms
+
+| Algorithm | Rust client |
+|-----------|-------------|
+| Hmac      | ❌         |
+| CbcMac    | ❌         |
+| Cmac      | ❌         |
```
Comment (sbailey-arm) [inline src/clients_api_coverage.md:117]: For things like these algorithms, where the algorithms _are_ supported, but the operations that use them are not, should they be ticks instead?

Comment (hug-dev) [reply src/clients_api_coverage.md:117]: I would say no personally. I think this page could be presented in a better way. The first table kind of assume that the operation is fully supported when we only really support one specific algorithm.
Ideally, we would have one massive matrix of combination of providers X combination of operations X combinations of their input parameters.

Comment (hug-dev) [reply src/clients_api_coverage.md:116]: And now that all single-part operations are defined, we could think of generating such table programatically, nightly, by either defining new operations that would give us the elvel of support for each operation OR by making test for each one of them. That could be a cool internship project 😉

**Comment by sbailey-arm** [comment]
This _should_ pass CI now the new protobuf contracts are in master.

**Review by hug-dev — APPROVED** [review]
Looks good to me, thank!
The link checker is palying us tricks. I think it's worth investigating why it fails, or using another one. Maybe the GitHub links do some kind of weird redirection that it does not support?
In any case, happy to merge this even if hte CI fails.

File: src/parsec_client/operations/psa_sign_message.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
+# PsaSignMessage
+
+Sign a message with a private key. Opcode: 23 (`0x0017`)
```
Comment (ionut-arm) [inline src/parsec_client/operations/psa_sign_message.md:3]: `PsaMacVerify` and this one have the same opcode!

File: src/service_api_coverage.md [hunk 5]
```diff
@@ -126,6 +131,27 @@ option marked as supported is necessarily supported for all operations on which
...
+### Cipher algorithms
+
+| Algorithm    | Mbed Crypto provider | PKCS 11 provider | TPM 2.0 provider |
+|--------------|----------------------|------------------|------------------|
+| StreamCipher | ❌                  | ❌              | ❌              |
+| Crt          | ❌                  | ❌              | ❌              |
+| Cfb          | ❌                  | ❌              | ❌              |
+| Ofb          | ❌                  | ❌              | ❌              |
+| Xts          | ❌                  | ❌              | ❌              |
+| EbcNoPadding | ❌                  | ❌              | ❌              |
```
Comment (ionut-arm) [inline src/service_api_coverage.md:143]:
```suggestion
| EcbNoPadding | ❌                  | ❌              | ❌              |
```

File: src/service_api_coverage.md [hunk 6]
```diff
@@ -126,6 +131,27 @@ option marked as supported is necessarily supported for all operations on which
 | GCM               | ✅                  | ❌              | ❌              |
 | ChaCha20_Poly1305 | ✅                  | ❌              | ❌              |
 
+### Cipher algorithms
+
+| Algorithm    | Mbed Crypto provider | PKCS 11 provider | TPM 2.0 provider |
+|--------------|----------------------|------------------|------------------|
+| StreamCipher | ❌                  | ❌              | ❌              |
+| Crt          | ❌                  | ❌              | ❌              |
```
Comment (ionut-arm) [inline src/service_api_coverage.md:139]:
```suggestion
| Ctr          | ❌                  | ❌              | ❌              |
```

File: src/service_api_coverage.md (hunk 5 above)
Comment (ionut-arm) [reply src/service_api_coverage.md:143]: This one and the one below also apply to the client coverage page

**Review by ionut-arm — COMMENTED** [review]
Thanks!! That is a lot of changes 😮

File: src/parsec_client/operations/psa_sign_message.md (hunk 4 above)
Comment (sbailey-arm) [reply src/parsec_client/operations/psa_sign_message.md]: 👀

**Review by ionut-arm — APPROVED** [review]

**Pull request merged**